src/exchange_adapters/mercado.ts

```diff
@@ -0,0 +1,114 @@
+import { Exchange } from '../utils'
+import { BaseExchangeAdapter, ExchangeAdapter, ExchangeDataType, Ticker, Trade } from './base'
+
+export class MercadoAdapter extends BaseExchangeAdapter implements ExchangeAdapter {
+  baseApiUrl = 'https://api.mercadobitcoin.net/api/v4'
+  readonly _exchangeName = Exchange.MERCADO
+  readonly _certFingerprint256 =
+    '3A:BB:E6:3D:AF:75:6C:50:16:B6:B8:5F:52:01:5F:D8:E8:AC:BE:27:7C:50:87:B1:27:A6:05:63:A8:41:ED:8A'
+
+  private static readonly tokenSymbolMap = MercadoAdapter.standardTokenSymbolMap
+
+  protected generatePairSymbol(): string {
+    const base = MercadoAdapter.tokenSymbolMap.get(this.config.baseCurrency)
+    const quote = MercadoAdapter.tokenSymbolMap.get(this.config.quoteCurrency)
+    return `${base}-${quote}`
+  }
+
+  async fetchTicker(): Promise<Ticker> {
+    const tickerJson = await this.fetchFromApi(
+      ExchangeDataType.TICKER,
+      `tickers?symbols=${this.pairSymbol}`
+    )
+    return this.parseTicker(tickerJson)
+  }
+
+  async fetchTrades(): Promise<Trade[]> {
+    // Trade data is not needed by oracle but is required by the parent class.
+    // This function along with all other functions that are not needed by the oracle will
+    // be removed in a future PR.
+    // -- @bayological ;) --
+    return []
+  }
+
+  /**
+   *
+   * @param json parsed response from mercado's ticker endpoint
+   * https://api.mercadobitcoin.net/api/v4/docs#tag/Public-Data/paths/~1tickers/get
+   * https://api.mercadobitcoin.net/api/v4/tickers?symbols=BTC-BRL
+   * [
+   *  {
+   *  "pair":"BTC-BRL",
+   *  "high":"120700.00000000",
+   *  "low":"117000.00001000",
+   *  "vol":"52.00314436",
+   *  "last":"119548.04744932",
+   *  "buy":"119457.96889001",
+   *  "sell":"119546.04397687",
+   *  "open":"119353.86994450",
+   *  "date":1674561363
+   *  }
+   * ]
+   *
+   */
+  // Using lastPrice to convert from baseVolume to quoteVolume, as Mercado's
+  // API does not provide this information. The correct price for the
+  // conversion would be the VWAP over the period contemplated by the ticker,
+  // but it's also not available. As a price has to be chosen for the
+  // conversion, and none of them are correct, lastPrice is chose as it
+  // was actually on one trade (whereas the buy or sell could have no
+  // relation to the VWAP). Similar argumentation for ask and bid.
+
```

Comment:
Neat problem-solving under constrained data! ❤️‍🔥 

The trade-off as I understand it would be between adding Mercado as an exchange diversification for sparse BRL data vs. corrupting the VWAP oracle rate with this approximation. Without looking at the data in detail, my gut feeling tells me it's still worth it, especially since there is around the same BTCBRL volume traded on Mercado as on Binance, another rate we use. 

One possible improvement I would argue for is not to use the last price to infer quote volume, but the mean of high and low price, which is I think a better approximation for VWAP. 

Proposal:
```
const highPrice = this.safeBigNumberParse(json[0].high)!
const lowPrice = this.safeBigNumberParse(json[0].low)!
const avgPrice = (highPrice.plus(lowPrice)).dividedBy(2) // pls double check in typescript this is really the mean
const quoteVolume = baseVolume?.multipliedBy(avgPrice)
```

@sissnad what do you think?

Reply:
Have a look at this [endpoint](https://www.mercadobitcoin.net/api/BTC/day-summary/2023/01/24/). Seems it gives both base and quote volume.
And to get best bid and ask I would use the order book endpoint.